packages/roosterjs-editor-api/lib/format/getFormatState.ts

```diff
@@ -49,6 +49,7 @@ export function getElementBasedFormatState(
         canUnlink: !!editor.queryElements('a[href]', QueryScope.OnSelection)[0],
         canAddImageAltText: !!editor.queryElements('img', QueryScope.OnSelection)[0],
         isBlockQuote: !!editor.queryElements('blockquote', QueryScope.OnSelection)[0],
+        isCode: !!editor.queryElements('code', QueryScope.OnSelection)[0],

```

Comment:
Should this be testing for the `pre` element rather than `code`? In other RTEs I've seen `pre` used for code blocks and `code` used for inline code.

Reply:
It depends on if you want to treat a pure PRE tag as code. It can happen if you copy from some content that contains PRE tag and paste, so there will be PRE but the content may not be code. But if you toggle code block from roosterjs, there will always be both PRE and CODE.

I personally fill both CODE and PRE is fine to me, so just depends on your requirement.

Reply:
I changed to check `PRE > CODE` instead